src/lib_json/json_reader.cpp

```diff
@@ -120,9 +120,9 @@ bool Reader::parse(std::istream& is, Value& root, bool collectComments) {
 
   // Since JSONCPP_STRING is reference-counted, this at least does not
   // create an extra copy.
-  JSONCPP_STRING doc;
-  std::getline(is, doc, (char)EOF);
-  return parse(doc.data(), doc.data() + doc.size(), root, collectComments);
+  std::getline(is, document_, (char)EOF);
+
+  return parse(document_.data(), document_.data() + document_.size(), root, collectComments);
```

Comment:
Are you sure this is needed?
```cpp
 106 bool Reader::parse(const std::string& document,
 107                    Value& root,
 108                    bool collectComments) {
 109   document_.assign(document.begin(), document.end());
 110   const char* begin = document_.c_str();
 111   const char* end = begin + document_.length();
 112   return parse(begin, end, root, collectComments);
 113 }
```
Doesn't that do the same thing? Are you just trying to avoid a copy? I don't see anything actually different here. The error-message points into `document_` either way, doesn't it?

* http://www.cplusplus.com/reference/string/string/assign/
* https://stackoverflow.com/questions/194634/string-c-str-vs-data

Reply:
Yep, you are right I just forgot to look through the methods and avoid changing the exact behavior of the method. No, I am not intended to avoid copy in this strange manner